src/binlog.cpp

```diff
@@ -101,10 +101,13 @@ int BinLog::write_msg(const struct buffer *buffer)
         source_component_id = msg->compid;
     }
 
-    /* set the expected system id to the first autopilot that we get a heartbeat from */
-    if (_target_system_id == -1 && msg_id == MAVLINK_MSG_ID_HEARTBEAT
-        && source_component_id == MAV_COMP_ID_AUTOPILOT1) {
-        _target_system_id = source_system_id;
+    /* set the expected system id to the first autopilot that we get a heartbeat from if fcu_id isn't settled */
+    if (_target_system_id == -1) {
```

Comment:
maybe easier to just set _target_system_id in the constructor looking at the config value we receive as parameter?

Reply:
done